include/roboptim/core/function/polynomial.hh

```diff
@@ -72,6 +72,16 @@ namespace roboptim
     value_type applyPolynomial
     (const_vector_ref coeffs, const_argument_ref x) const;
 
+  public:
+    /// \brief Get the type-checking flag
+    virtual unsigned long flag()
+    {
+      return flag_;
+    }
+
+    /// \brief Flag representing the Roboptim Function type
+    static const unsigned long flag_;
+
```

Comment:
Yup, I was wondering about putting it in one of the `ROBOPTIM_FUNCTION_FWD_` lookalike macro, but I think the solution now is maybe a little more readable, even if it implies a copy/paste in all of the function-types headers.
For the typedef thing, let's mix it with the `enum`.